### PR TITLE
Add currentUrl accessor to WebView plugin.

### PR DESCRIPTION
**Pull request by efortuna**
This allows Flutter to query to WebView to find out what URL it is currently displaying.

### CODE REVIEW

File: packages/webview_flutter/test/webview_flutter_test.dart [hunk 1]
```diff
@@ -254,19 +254,49 @@ void main() {
...
+        },
+      ),
+    );
+
+    expect(controller, isNotNull);
+
+    // Test a WebView without an explicitly set first URL.
+    expect(await controller.currentUrl(), null);
+
+    final FakePlatformWebView platformWebView =
```
Comment (efortuna) [inline packages/webview_flutter/test/webview_flutter_test.dart]: Please note: I was a little uncertain how these tests were actually done with FakePlatformWebView. Is it possible that by me adding the currentUrl accessor, that FakePlatformWebView is no longer needed? If so this part of the code will need to be changed.

Comment (amirh) [reply packages/webview_flutter/test/webview_flutter_test.dart]: Yes it's actually nice that we can use the public API for testing here.
We still need the `FakePlatformWebView` implementation so something responds to the method channel.
But yes, in this test (and the other tests that only used the `FakePlatformWebView` directly to get the current url) we don't need to fetch `fakePlatformViewsController.lastCreatedView` any more.

Sweet 😄

Comment (amirh) [reply packages/webview_flutter/test/webview_flutter_test.dart]: Actually if we're migrating the other tests that used `platformWebView.currentUrl` to instead do `controller.currentUrl()` we probably don't need a 'Current URL' test.

File: packages/webview_flutter/test/webview_flutter_test.dart [hunk 2]
```diff
@@ -290,7 +320,7 @@ class FakePlatformWebView {
   List<String> history = <String>[];
   int currentPosition = -1;
 
-  String get currentUrl => history.isEmpty ? null : history[currentPosition];
+  String get url => history.isEmpty ? null : history[currentPosition];
```
Comment (amirh) [inline packages/webview_flutter/test/webview_flutter_test.dart]: Curious why did you prefer to rename this to `url`? (especially as now this is what we return for 'currentUrl' method channel invocations)

**Comment by amirh** [comment]
We also need to make the bots happy 😄

File: packages/webview_flutter/test/webview_flutter_test.dart (hunk 1 above)
Comment (efortuna) [reply packages/webview_flutter/test/webview_flutter_test.dart]: updated!

File: packages/webview_flutter/lib/webview_flutter.dart [hunk 3]
```diff
@@ -207,9 +207,13 @@ class WebViewController {
 
   /// Accessor to the current URL that the WebView is displaying.
   ///
-  /// If initialUrl was never specified, currentUrl returns `null`.
+  /// If [WebView.initialUrl] was never specified, or if , currentUrl returns `null`.
```
Comment (amirh) [inline packages/webview_flutter/lib/webview_flutter.dart]: uber nit: extra space before comma (`...or if , currentUrl...`)

I'm not sure what "if currentUrl returns `null`" means... you mean if the method channel call returns null ?
I feel like we should try to make the documentation here be all the user needs to read. I'd document any relevant platform-specific behavior here, and treat the method channel API as a private implementation detail as much as possible.

Comment (efortuna) [reply packages/webview_flutter/lib/webview_flutter.dart]: oops that was a text-editing fail on my part. I meant "If [WebView.initialUrl] was never specified, returns `null`." Fixed!

**Pull request merged**